xml/admin_saltcluster.xml

```diff
@@ -1376,6 +1376,231 @@ OSD_ID  HOST         STATE                    PG_COUNT  REPLACE  FORCE  STARTED_
+     </para>
+     <substeps>
+      <step>
+       <para>
+        Create the necessary directories:
+       </para>
+<screen>&prompt.root;mkdir -p /var/lib/registry/{auth,certs}</screen>
+      </step>
+      <step>
+       <para>
+        Generate an SSL certificate:
+       </para>
+<screen>
+&prompt.root;openssl req -newkey rsa:4096 -nodes -sha256 \
+ -keyout /var/lib/registry/certs/domain.key -x509 -days 365 \
+ -out /var/lib/registry/certs/domain.crt
+</screen>
+       <note>
+        <para>
+         Set the <literal>CN=[value]</literal> value to the fully qualified
+         domain name of the host ([<option>REQ_HOST_FQDN</option>]).
+        </para>
+       </note>
+      </step>
+      <step>
+       <para>
+        Copy the certificate to all cluster nodes and refresh the certificate
+        cache:
+       </para>
+<screen>
+&prompt.root;salt-cp '*' /var/lib/registry/certs/domain.crt \
+ /etc/pki/trust/anchors/
+&prompt.root;salt '*' cmd.shell "update-ca-certificates"
+</screen>
+      </step>
+      <step>
+       <para>
+        Generate a username and password combination for authentication to the
+        registry:
+       </para>
+<screen>
+&prompt.root;htpasswd2 -bBc /var/lib/registry/auth/htpasswd \
+ <option>REG_USERNAME</option> <option>REG_PASSWORD</option>
+</screen>
+      </step>
+      <step>
+       <para>
+        Start the secure refistry:
+       </para>
+<screen>
+podman run --name myregistry -p <option>REG_HOST_PORT</option>:5000 \
+ -v /var/lib/registry:/var/lib/registry \
+ -v /var/lib/registry/auth:/auth:z \
+ -e "REGISTRY_AUTH=htpasswd" \
+ -e "REGISTRY_AUTH_HTPASSWD_REALM=Registry Realm" \
+ -e REGISTRY_AUTH_HTPASSWD_PATH=/auth/htpasswd \
+ -v /var/lib/registry/certs:/certs:z \
+ -e "REGISTRY_HTTP_TLS_CERTIFICATE=/certs/domain.crt" \
+ -e "REGISTRY_HTTP_TLS_KEY=/certs/domain.key" \
+ -e REGISTRY_COMPATIBILITY_SCHEMA1_ENABLED=true -d registry:2
+</screen>
+      </step>
+      <step>
+       <para>
+        Test secure access to the registry:
+       </para>
+<screen>
+&prompt.user;curl https://<option>REG_HOST_FQDN</option>:<option>REG_HOST_PORT</option>/v2/_catalog \
+ -u <option>REG_USERNAME</option>:<option>REG_PASSWORD</option>
+</screen>
+      </step>
+      <step>
+       <para>
+        Test podman access to the registry:
+       </para>
+<screen>
+&prompt.user;podman login <option>REG_HOST_FQDN</option>:<option>REG_HOST_PORT</option>
+# Enter <option>REG_USERNAME</option> and <option>REG_PASSWORD</option> when prompted
+&prompt.user;podman inspect <option>HOST_FQDN</option>/<option>IMAGE</option>:<option>TAG</option>
+&prompt.user;podman logout <option>HOST_FQDN</option>
+</screen>
+      </step>
+      <step>
+       <para>
+        Enable a secure &cephadm; connection:
+       </para>
+<screen>
+cephadm registry-login \
+ --registry-url <option>HOST_FQDN</option>:<option>HOST_PORT</option> \
+ --registry-username <option>REG_USERNAME</option> \
+ --registry-password <option>REG_PASSWORD</option>
```

Comment:
You probably want to do that before bootstrapping the cluster.

Reply:
> You probably want to do that before bootstrapping the cluster.

you mean i should insert a reference to this section from the deployment process for admins that want to use local registry?

Reply:
I don't think you need to include the `cephadm registry-login` command here. `ceph-salt apply` will (or should) take care of doing that, provided you've specified the registry_auth config as mentioned in [5.3.2.11 Configuring the container registry](https://documentation.suse.com/ses/7/single-html/ses-deployment/index.html#deploy-cephadm-configure-registry)